src/utils/hash.ts

```diff
@@ -147,6 +147,38 @@ export function calculateDeployTransactionHash(
   );
 }
 
+export function calculateDeclareTransactionHash(
+  // contractClass: ContractClass, // Should be used once class hash is present in ContractClass
+  classHash: BigNumberish,
+  senderAddress: BigNumberish,
+  version: BigNumberish,
+  maxFee: BigNumberish,
+  chainId: StarknetChainId,
+  nonce: BigNumberish
+): string {
+  let calldata: BigNumberish[] = [];
+  let additionalData: BigNumberish[] = [];
+
+  if (version !== ZERO) {
```

Comment:
as this function is new, do we really wanna keep support for old versions?
I would just assume version 1 and remove version 0 logic.

Also keep in mind estimating fees has a different version too!

(Also I'm not sure native operations like `!==` work on non-native BigInts, do they?)